lnwallet/channel.go

```diff
@@ -5553,7 +5553,10 @@ func genHtlcSigValidationJobs(chanState *channeldb.OpenChannel,
 			// disk later.
 			sigType := htlcCustomSigType.TypeVal()
 			auxSig.WhenSome(func(sigB tlv.Blob) {
```

Comment:
This panic behavior was exposed on this [litd itest](https://github.com/lightninglabs/lightning-terminal/pull/842)

Not sure if I should do the nil check, as is now, or if I should be instead correctly initializing the map.

This current diff seems to be satisfying for the itest linked above

Reply:
I think we should initialize the map correctly instead. Interestingly I ran into the same thing in part 3 of the rebased series today. I did the initialization on nil there.

Reply:
This is how it looks in master now:

```
			if htlc.CustomRecords == nil {
				htlc.CustomRecords = make(lnwire.CustomRecords)
			}
```

See https://github.com/lightningnetwork/lnd/commit/83fdbda2fa9f9c2fec54855d34ea7eb65911d571